transaction-status/src/parse_token/extension/token_group.rs

```diff
@@ -0,0 +1,209 @@
+use {
+    super::*,
+    spl_token_group_interface::instruction::{
+        InitializeGroup, TokenGroupInstruction, UpdateGroupAuthority, UpdateGroupMaxSize,
+    },
+};
+
+pub(in crate::parse_token) fn parse_token_group_instruction(
+    instruction: &TokenGroupInstruction,
+    account_indexes: &[u8],
+    account_keys: &AccountKeys,
+) -> Result<ParsedInstructionEnum, ParseInstructionError> {
+    match instruction {
+        TokenGroupInstruction::InitializeGroup(group) => {
+            check_num_token_accounts(account_indexes, 3)?;
+            let InitializeGroup {
+                max_size,
+                update_authority,
+            } = group;
+            let mut value = json!({
+                "group": account_keys[account_indexes[0] as usize].to_string(),
+                "maxSize": u32::from(*max_size),
+                "mint": account_keys[account_indexes[1] as usize].to_string(),
+                "mintAuthority": account_keys[account_indexes[2] as usize].to_string(),
+            });
+            let map = value.as_object_mut().unwrap();
+            if let Some(update_authority) = Option::<Pubkey>::from(*update_authority) {
+                map.insert(
+                    "updateAuthority".to_string(),
+                    json!(update_authority.to_string()),
+                );
+            }
+            Ok(ParsedInstructionEnum {
+                instruction_type: "initializeTokenGroup".to_string(),
+                info: value,
+            })
+        }
+        TokenGroupInstruction::UpdateGroupMaxSize(update) => {
+            check_num_token_accounts(account_indexes, 2)?;
+            let UpdateGroupMaxSize { max_size } = update;
+            let value = json!({
+                "group": account_keys[account_indexes[0] as usize].to_string(),
+                "maxSize": u32::from(*max_size),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+            });
+            Ok(ParsedInstructionEnum {
+                instruction_type: "updateTokenGroupMaxSize".to_string(),
+                info: value,
+            })
+        }
+        TokenGroupInstruction::UpdateGroupAuthority(update) => {
+            check_num_token_accounts(account_indexes, 2)?;
+            let UpdateGroupAuthority { new_authority } = update;
+            let mut value = json!({
+                "group": account_keys[account_indexes[0] as usize].to_string(),
+                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
+            });
+            let map = value.as_object_mut().unwrap();
+            if let Some(new_authority) = Option::<Pubkey>::from(*new_authority) {
+                map.insert("newAuthority".to_string(), json!(new_authority.to_string()));
+            }
```

Comment:
Can the new authority be `None` if it's not being set? That's how `SetAuthority` is handling it, so we should be able to simplify this to:
```suggestion
            let value = json!({
                "group": account_keys[account_indexes[0] as usize].to_string(),
                "updateAuthority": account_keys[account_indexes[1] as usize].to_string(),
                "newAuthority": Option::<Pubkey>::from(*new_authority).map(|v| v.to_string())
            });
```